pkg/controller/machine.go

```diff
@@ -404,13 +404,24 @@ func (c *Controller) syncHandler(key string) error {
 					return fmt.Errorf("failed to create bootstrap kubeconfig: %v", err)
 				}
 
-				data, err := userdataProvider.UserData(machine.Spec, kubeconfig, prov, c.clusterDNSIPs)
+				//Need the kubeconfig to extract the CACert for the userdata rendering
+				cm, err := c.kubeconfigProvider.GetKubeconfig()
 				if err != nil {
-					return fmt.Errorf("failed get userdata: %v", err)
+					return fmt.Errorf("failed to get kubeconfig: '%v'", err)
+				}
+				if len(cm.Clusters) != 1 {
+					return fmt.Errorf("kubeconfig does not contain exactly one cluster, can not extract cacert...")
+				}
+				var userData string
+				for _, clusterConfig := range cm.Clusters {
+					userData, err = userdataProvider.UserData(machine.Spec, kubeconfig, prov, c.clusterDNSIPs, string(clusterConfig.CertificateAuthorityData))
+					if err != nil {
+						return fmt.Errorf("failed get userdata: %v", err)
+					}
```

Comment:
I would rather implement a function in the KubeconfigProvider.

Reply:
Not sure I understand that, a function in the KubeConfigProvider to get the userdata..?

Reply:
Fetching of the CA data should be implemented as a function in the KubeconfigProvider